[client] createDeployment failing "MODULE_NOT_FOUND"

### Discussed in https://github.com/orgs/vercel/discussions/8466

<div type='discussions-op-text'>

<sup>Originally posted by **eric-miller2129** August 25, 2022</sup>
Hi there, 

I'm using `@vercel/client` to run a programmatic deployment to Vercel. Until recently it was working fine however it seems once I bumped the version from `12.1.11` to `12.2.0` it began to fail.

![Screen Shot 2022-08-25 at 17 38 57](https://user-images.githubusercontent.com/1424187/186773899-b9369942-df90-41a6-a486-80015b71c00d.png)

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'tar-fs'
Require stack:
- .../node_modules/@vercel/client/dist/create-deployment.js
- .../node_modules/@vercel/client/dist/index.js
```
Switching back to `12.1.11` solves the issue, and just manually installing `tar-fs` also seems to solve the issue, but I don't think either of those things are the correct solution.

---

Going through the commit history, it looks like this may be coming from the addition of `tar-fs` here https://github.com/vercel/vercel/blob/603b1256c63e46ad54da28aa7963d737349bec4d/packages/client/src/create-deployment.ts#L13

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -32,7 +32,6 @@
     "@types/node-fetch": "2.5.4",
     "@types/recursive-readdir": "2.2.0",
     "@types/tar-fs": "1.16.1",
-    "tar-fs": "1.16.3",
     "typescript": "4.3.4"
   },
   "jest": {
@@ -55,6 +54,7 @@
     "ms": "2.1.2",
     "node-fetch": "2.6.7",
     "querystring": "^0.2.0",
-    "sleep-promise": "8.0.1"
+    "sleep-promise": "8.0.1",
+    "tar-fs": "1.16.3"
   }
 }

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -31,7 +31,8 @@
     "@types/node": "12.0.4",
     "@types/node-fetch": "2.5.4",
     "@types/recursive-readdir": "2.2.0",
-    "@types/tar-fs": "^2.0.1",
+    "@types/tar-fs": "1.16.1",
+    "tar-fs": "1.16.3",
     "typescript": "4.3.4"
   },
   "jest": {

